### PR TITLE
Add functional autograd API: jacobian

### PR DESCRIPTION
**Pull request by levi131**
### PR types
New features

### PR changes
APIs

### Describe
Add functional autograd API: jacobian.
The API signature is:
     `def jacobian(func, inputs, create_graph=False, allow_unused=False)`

This API computes the Jacobian matrix of `func` with respect to `inputs`.
This API is ONLY available in imperative mode.

 Parameters:

        func (function): a Python function that takes a Tensor or a Tensor
            list/tuple as inputs and returns a Tensor or a Tensor tuple.
        inputs (Tensor|list(Tensor)|tuple(Tensor)): the input Tensor or 
            Tensor list/tuple of the function ``func``.
        create_graph (bool, optional): whether to create the gradient graphs
            of the computing process. When it is True, higher order derivatives
            are supported to compute; when it is False, the gradient graphs of
            the computing process would be discarded. Defaults to ``False``.
        allow_unused (bool, optional): whether to raise error or return None if
            some Tensors of `inputs` are unreachable in the graph. Error would
            be raised if allow_unused=False, and None would be returned as
            their gradients if allow_unused=True. Default False.

 Returns:

        Jacobian (Tensor or nested tuple of Tensors): if function ``func``
        takes a Tensor as inputs and returns a Tensor as outputs, Jacobian
        will be a single Tensor containing the Jacobian matrix for the
        linearized inputs and outputs. If one of the inputs and outputs is
        a Tensor, and another is a Tensor list/tuple, then the Jacobian will
        be a tuple of Tensors. If both of inputs and outputs are Tensor
        list/tuple, then the Jacobian will be a tuple of tuple of Tensors
        where ``Jacobian[i][j]`` will contain the Jacobian matrix of the
        linearized ``i``th output and ``j``th input and will have same
        dtype and device as the corresponding input. ``Jacobian[i][j]`` will
        have as size ``m * n``, where ``m`` and ``n`` denote the numbers of
        elements of ``i``th output and ``j``th input respectively.

TODO:
        1、We find a bug about `grad` interface when running testcase TestJacobianFloat64::test_create_graph_true, will fix it later.
        2、May add some pre-process on `inputs` to support jacobian API when inputs.stop_gradients = True.
        3、Vectorize the computing process.

### CODE REVIEW

**Review by JiabinYang — APPROVED** [review]
LGTM

File: python/paddle/autograd/functional.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    return paddle.stack(
+        origin_list, axis=0) if isinstance(origin_list[0],
+                                           paddle.Tensor) else None
+
+
+@framework.dygraph_only
+def jacobian(func, inputs, create_graph=False, allow_unused=False):
+    ''' 
+    .. note::
+        **This API is ONLY available in Dygraph mode.**
```
Comment (phlrain) [inline python/paddle/autograd/functional.py]: imperative mode

**Review by JiabinYang — CHANGES_REQUESTED** [review]
Fix bugs first

File: python/paddle/autograd/functional.py (hunk 1 above)
Comment (levi131) [reply python/paddle/autograd/functional.py]: done

**Review by XieYunshen — DISMISSED** [review]
LGTM for set_tests_properties(test_jacobian PROPERTIES TIMEOUT 20)

File: python/paddle/autograd/functional.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        linearized inputs and outputs. If one of the inputs and outputs is
+        a Tensor, and another is a Tensor list/tuple, then the Jacobian will
+        be a tuple of Tensors. If both of inputs and outputs are Tensor
+        list/tuple, then the Jacobian will be a tuple of tuple of Tensors
+        where ``Jacobian[i][j]`` will contain the Jacobian matrix of the
+        linearized ``i``th output and ``j``th input and will have same
+        dtype and device as the corresponding input. ``Jacobian[i][j]`` will
+        have as size ``m * n``, where ``m`` and ``n`` denote the numbers of
+        elements of ``i``th output and ``j``th input respectively.
+    '''
```
Comment (TCChenlong) [inline python/paddle/autograd/functional.py:153]: 缺少示例代码

**Review by TCChenlong — DISMISSED** [review]

File: python/paddle/autograd/functional.py (hunk 2 above)
Comment (levi131) [reply python/paddle/autograd/functional.py:153]: 已经补充

**Review by phlrain — APPROVED** [review]

**Review by CheQiXiao — APPROVED** [review]

**Review by XieYunshen — APPROVED** [review]

**Review by PangHua — APPROVED** [review]

**Review by JiabinYang — DISMISSED** [review]
LGTM

**Review by TCChenlong — APPROVED** [review]

**Pull request merged**